core/src/main/java/com/linecorp/armeria/client/ClientBuilder.java

```diff
@@ -84,6 +159,38 @@ public ClientBuilder factory(ClientFactory factory) {
      */
     public <T> T build(Class<T> clientType) {
         requireNonNull(clientType, "clientType");
-        return factory.newClient(uri, clientType, buildOptions());
+        checkArguments();
+
+        if (uri != null) {
+            return factory.newClient(uri, clientType, buildOptions());
+        } else if (path != null) {
+            return factory.newClient(scheme(), endpoint, path, clientType, buildOptions());
+        } else {
+            return factory.newClient(scheme(), endpoint, clientType, buildOptions());
+        }
+    }
+
+    private Scheme scheme() {
+        return scheme == null ? Scheme.of(format, requireNonNull(protocol)) : scheme;
```

Comment:
Probably we can remove the null check for the `protocol` because when the `scheme` is `null`, the `protocol` is always not `null`?

Reply:
Yes, I think we do not need `requireNonNull` at here.